utils/cache.js

```diff
@@ -89,23 +161,65 @@ const cache = (data = { priority: 2, expiry: CACHE_EXPIRY_TIME_MIN }) => {
 
         res.send = (body) => {
           const cacheValue = {
-            priority: data.priority,
+            priority: priority,
             response: body,
-            expiry: new Date().getTime() + minutesToMilliseconds(data.expiry),
+            expiry: new Date().getTime() + minutesToMilliseconds(expiry),
             size: Buffer.byteLength(body),
           };
           pool.set(key, cacheValue);
+          if (modelKey) {
+            cachedKeys.addCachedKey(modelKey, key);
+          }
           res.send = oldSend;
           return res.send(body);
         };
 
         next();
       }
     } catch (err) {
-      logger.error(`Error while getting cached tasks response ${err}`);
+      logger.error(`Error while getting cached response ${err}`);
+      next();
+    }
+  };
+};
+
+/**
+ * Generate key to save cache.
+ * @param {Object} [request]  HTTP request argument to the middleware function passed by express.
+ * @returns {string} cache key.
+ */
+const generateCacheKey = (request) => {
+  return "__cache__" + request.method + request.originalUrl;
```

Comment:
Suggestion: As per key naming convention in Redis we use `:` as a separator. I think we can use the same as it will help make the code more readable and make the transition easier if in the future we decide to use Redis 

Reference - https://redis.io/docs/manual/keyspace/

Reply:
made changes to try follow the conventions mentioned there as close as possible